Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,103 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+/// An `IteratorProtocol` that can iterate through comma separated list of values for a certain
+/// header.
+///
+/// **Example:**
+///
+/// Suppose you have these headers:
+///
+///      Connection: keep-alive, x-server
+///      Content-Type: text/html
+///      Connection: other
+///
+/// You can iterate using this struct on those headers, for values of `Connection`, to get
+/// `keep-alive`, then `x-server`, then `other`
+public struct HTTPListHeaderIterator<Name: Collection>: Sequence, IteratorProtocol where Name.Element == UInt8 {
```

Comment:
that's probably a stupid question but why is this generic on the http header `Name`?

Reply:
To open the possibility to create it with something other than `Array<UInt8>`
You can use `ContiguousArray<UInt8>` (for performance), or even some custom linked list type if you want.

Reply:
@weissi This is becoming a somewhat common pattern in NIO and associated libraries to deal with the fact that we're now storing HTTP headers as `ByteBuffer`s instead of as `String`s: if you can provide us something that's a `ContiguousCollection` then we can do a faster comparison than if you give us something that's not (e.g. `String.UTF8View`), so we tend to make these things generic and let the user give us the best thing they can.

Reply:
@Lukasa do we really have another `public` API that is generic like that? I think this comes with a performance overhead (haven't tested this case yet) and I don't understand where we actually use this today. AFAIK currently all public API lets you only access the headers as a `String` which currently comes at a cost too but we'd hope for future Swift versions to fix that. Swift 4.2 is already an improvement here but there's much more that can be done like being able to access the contiguous ASCII representation of a `String` if available.

Reply:
@fadi-botros understand that but I'm interested in where we actually use that today. I'm pretty sure a generic public type comes with performance overhead at least when used from outside the module (which is most of the time). So I think we shouldn't make stuff generic just because we can.